### PR TITLE
Marker render CSS optimization

### PR DESCRIPTION
**Pull request by rshtg**
This kind of optimization gives us about 50-60 FPS on one zoom level

### CODE REVIEW

**Comment by Pessimistress** [comment]
@rshtg Thanks! Can you fix the lint errors https://travis-ci.org/uber/react-map-gl/jobs/287001934

File: src/components/marker.js [hunk 1]
```diff
@@ -53,14 +53,25 @@ export default class Marker extends BaseControl {
...
       top: y + offsetTop
     };
 
+    const isVisible = !(top < 0 || left < 0);
+
+    let containerStyle = {};
+
+    if (isVisible) {
+      containerStyle = {
+        transform: `translate3d(${left}px, ${top}px, 0)`
```
Comment (Pessimistress) [inline src/components/marker.js]: Why is CSS transform preferable than absolute positioning? Is it contributing to perf at all?

File: src/components/marker.js [hunk 2]
```diff
@@ -53,14 +53,25 @@ export default class Marker extends BaseControl {
...
+      };
+    }
+
     return createElement('div', {
-      className: `mapboxgl-marker ${className}`,
+      // isVisible toggles the .visible to show/hide the element
+      className: `mapboxgl-marker ${className} ${isVisible ? 'visible' : ''}`,
```
Comment (Pessimistress) [inline src/components/marker.js]: I can see that avoid rendering out-of-view markers provide perf benefit. Though I'm not convinced about the `visible` class. It adds an additional catch for the user to implement, and it doesn't account for the size of the marker.  @abmai

File: src/components/marker.js (hunk 1 above)
Comment (rshtg) [reply src/components/marker.js]: CSS transform is more productive property which uses GPU across rendering. I've complete some test with 50-100-150 markers on Mobile and it shows 2x better performance iOS Safari (Latest) / Android Chrome (Latest)

File: src/components/marker.js (hunk 2 above)
Comment (rshtg) [reply src/components/marker.js]: Am I need to put the marker sizes check. Or it would be better to roll it back?

Comment (Pessimistress) [reply src/components/marker.js]: @rshtg We don't ask the user to specify the size of each marker today. Dynamically querying their sizes may cause perf problems. Shall we remove it for now to land the transform change?

File: src/components/marker.js (hunk 1 above)
Comment (Pessimistress) [reply src/components/marker.js]: Shall we apply this technique to the Popup component as well?

File: src/components/marker.js (hunk 2 above)
Comment (rshtg) [reply src/components/marker.js]: I think we just can keep only transformation without `visible` toggler. So the profit should be affected in any case

File: src/components/marker.js (hunk 1 above)
Comment (rshtg) [reply src/components/marker.js]: Yes, I think each component should be transformed as well

**Comment by istarkov** [comment]
css transform can cause a lot of issues with hovering, 
visual issues like font smoothing,
and working with 3rd party components like tooltips.

May be this has been changed last 1-2 years, but in my library https://github.com/istarkov/google-map-react I also used style positioning and not because I don't have a knowledge about transforms and performance. 
Also @rshtg have you tried to use style transform with additional css on marker `will-change: left, top;` it must give the same effect on modern browsers as css transform.

**Pull request closed without merging**